src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -288,9 +287,8 @@ private void leaveLiteralNew(DetailAST ast) {
      */
     private void traverseFieldFrameTree(FieldFrame frame) {
         for (FieldFrame child: frame.getChildren()) {
-            if (!child.getChildren().isEmpty()) {
```

Comment:
This just prevents calling the same method when we know the iterator in the new method call will be empty and won't do anything.